src/tribler/core/libtorrent/restapi/torrentinfo_endpoint.py

```diff
@@ -219,3 +220,31 @@ async def get_torrent_info(self, request: Request) -> RESTResponse:  # noqa: C90
 
         return RESTResponse({"metainfo": hexlify(json_dump.encode()).decode(),
                              "download_exists": download and not download_is_metainfo_request})
+
+    @docs(
+        tags=["Libtorrent"],
+        summary="Return metainfo from a torrent found at a provided .torrent file.",
+        responses={
+            200: {
+                "description": "Return a hex-encoded json-encoded string with torrent metainfo",
+                "schema": schema(GetMetainfoResponse={"metainfo": String})
+            }
+        }
+    )
+    async def get_torrent_info_from_file(self, request: web.Request) -> RESTResponse:
+        """
+        Return metainfo from a torrent found at a provided .torrent file.
+        """
+        tdef = TorrentDef.load_from_memory(await request.read())
+        infohash = tdef.get_infohash()
+
+        # Check if the torrent is already in the downloads
+        download = self.download_manager.downloads.get(infohash)
+        metainfo_request = self.download_manager.metainfo_requests.get(infohash, [None])[0]
+        requesting_metainfo = download == metainfo_request
```

Comment:
This look similar to https://github.com/qstokkink/TriblerExperimental/pull/74/files#diff-40175f3551408276c12bea66ac0ed6672b8290e87d1fa1c8f483c04211f1e623R210-R212 Could you double-check if this needs the same fix?

Reply:
Oops, I fixed it on the main branch, but not on my own branch.